src/main.c

```diff
@@ -8,174 +8,184 @@
 #include <unistd.h>
 #include "built-in.h"
 
-#define MAXLINE 1000
+#define MAXLINE 1000 // buffer size
+#define MAXPROMPT 201 // 200 characters + nul
 
-void main_loop(void);
-int get_tokens(char *input, char *delim, char *args[], int *reset);
-void fork_child(char *args[]);
+static void main_loop(void);
+static int get_tokens(char *input, bool *reset);
+static void fork_child(char *args[], bool background);
+static char *args[MAXLINE];
 
 int main(void) {
-
     using_history();
-
-    main_loop();    /* Enter the main event loop */
-
-    return 0;
+    main_loop();
 }
 
 void main_loop(void) {
-    char *input;            /* String to store input */
-    char *args[MAXLINE];            /* Array of strings to store all arguments */
-    int i, len;
-    int background_process = 0;     /* Flag to check if the process is to be run in the background */
-    char credentials[MAXLINE] = "@";
-    int reset_val = 0;
-    int *reset = &reset_val;
-
-    system("clear");                /* Clear the screen on launch */
+    system("clear");
     
-    while (1) {                     /* Infinite loop */
+    while (1) {
+        // Create PS: "@{login}-->{cwd} > "
+        char const * login = getlogin();
+        char const * cwd = getcwd(NULL, 0);
+        char prompt_string[MAXLINE];
+        prompt_string[0] = '\0';
+        if (login && cwd) {
+            prompt_string[0] = '@';
+            prompt_string[1] = '\0';
+            strlcat(prompt_string, login, MAXPROMPT);
+            strlcat(prompt_string, "-->", MAXPROMPT);
+            strlcat(prompt_string, cwd, MAXPROMPT);
+        }
+        free((void *) cwd);
+        strlcat(prompt_string, " > ", MAXPROMPT);
 
-        /* Gets the username and the current working directory to display as the prompt */
-        credentials[1] = '\0';
-        strcat(credentials, getlogin());
-        strcat(credentials, "-->");
-        strcat(credentials, getcwd(NULL, 0));
-        strcat(credentials, " > ");
 
-        /* Display the prompt and add the user's input to history */
-        input = readline(credentials);
+        // Display the prompt and add the user's input to history
+        char *input = readline(prompt_string);
+        if (!input) // EOF
+            exit(0);
         add_history(input);
 
-        len = strlen(input);
-
-        if (input[len - 1] == '\n')
-            input[len - 1] = '\0';                  /* Replace the newline with a nul terminator */
-
-        i = get_tokens(input, " ", args, reset);   /* Gets all tokens from input */
-
-        if (i == -1) {      /* Continue if get_tokens failed*/
-            continue;
-        }
+        // TODO
+        /* // strip newline from end */
+        /* int len = strlen(input); */
+        /* if (input[len - 1] == '\n') */
+        /*     input[len - 1] = '\0'; */
```

Comment:
Didn't want to delete it because I thought there probably is a reason it was there in the first place. I wasn't able to spot anything wrong when commenting it out though...